```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
rm(list = ls())
```

Import metadata:
```{r}
library(stringr)

# metadata <- read.csv('/mnt/synology/GERARD/DATA/METAHIT/DNA/samples_list_metadata_with_dotsinID.tsv', sep = "\t", header = T, row.names = 7) #METAHIT ORIGINALS
#
metadata <- read.csv('/mnt/synology/GERARD/DATA/METAHIT/DNA_JOINED/samples_list_metadata_with_dotsinID.tsv', sep = "\t", header = T, row.names = 7) #METAHIT JOINED
# metadata <- read.csv('E:/JULIOL/DATA/METAHIT/DNA_JOINED/samples_list_metadata_with_dotsinID.tsv', sep = "\t", header = T, row.names = 7) #METAHIT JOINED4
# metadata <- read.csv('C:/Users/Gerard/Downloads/JULIOL/DATA/METAHIT/DNA_JOINED/samples_list_metadata_with_dotsinID.tsv', sep = "\t", header = T, row.names = 7) #METAHIT JOINED4

patient <- paste(str_split(rownames(metadata), pattern = '[.]', simplify = T)[,2], str_split(rownames(metadata), pattern = '[.]', simplify = T)[,4], sep = "" )
timep <- str_split(rownames(metadata), pattern = '[.]', simplify = T)[,3]
patient <- substr(patient, 3, nchar(patient))
metadata <- cbind(metadata, patient, timep)

rm(patient)

metadata$remission_vs_recurrence[metadata$remission_vs_recurrence == "RELAPSE"] <- "Relapse"
metadata$remission_vs_recurrence[metadata$remission_vs_recurrence == "REMISSION"] <- "Remission"

metadata$cohort_UC[metadata$cohort_UC == "healthy"] <- "Healthy"

metadata$remission_vs_recurrence <- as.factor(metadata$remission_vs_recurrence)


metadata$remission_vs_recurrence <- as.factor(metadata$remission_vs_recurrence)
metadata$origin <- "Europe"
metadata$country <- "Spain"
metadata$cohort_UC[metadata$cohort_UC == "HR"] <- "Healthy"
```

<!-- Merge the bug_list (Metaphlan output) into 2 data frames. bug_list is a dataframe with just the species detected. additional_species is data frame with the species detected the NCBI ID and the additional species per sample. -->
<!-- ```{r, include=FALSE} -->
<!-- library(readr) -->

<!-- # wd <- '/mnt/synology/GERARD/HUMANN3_RUNS/KATHLEEN_SAMPLES/' #workind directory where output folders are -->

<!-- wd <- '/mnt/synology/GERARD/HUMANN3_RUNS/METAHIT_DNA_RERUN/' #METAHIT RERUN -->

<!-- wd <- '/mnt/synology/GERARD/HUMANN3_RUNS/METAHIT_DNA_ORIGINAL/' #METAHIT ORIGINALS -->

<!-- wd <- '/mnt/synology/GERARD/HUMANN3_RUNS/METAHIT_DNA_JOINED/' #METAHIT JOINED -->

<!-- setwd(wd) -->

<!-- counter <- 1 -->

<!-- for( samp in list.files()) { -->
<!--   filepath <- paste(wd , samp ,'/', samp , '_combined_reads_metaphlan_bugs_list.tsv', sep = '') -->
<!--   if (samp != 'logs') { -->
<!--     if (counter == 1) { -->
<!--     bug_list <- read_delim(file = filepath, "\t",  col_names = FALSE, comment = "#", escape_double = FALSE, trim_ws = TRUE) -->
<!--     additional_species <- data.frame(bug_list[,1:2], bug_list[,4:4]) -->
<!--     bug_list <- data.frame(bug_list[,c(1,3)]) -->
<!--     counter <- counter + 1 -->
<!--     colnames(additional_species)[counter+1] <- paste('additional_species_',samp,sep='') -->
<!--     colnames(bug_list)[counter] <- samp -->

<!--     } -->
<!--     else { -->
<!--     bug_list_temp <- read_delim(file = filepath, "\t",  col_names = FALSE, comment = "#", escape_double = FALSE, trim_ws = TRUE) -->
<!--     additional_species_temp <- data.frame(bug_list_temp[,1:2], bug_list_temp[,4:4]) -->
<!--     bug_list_temp <- data.frame(bug_list_temp[,c(1,3)]) -->
<!--     bug_list <- merge(bug_list, bug_list_temp, all = TRUE) -->
<!--     additional_species <- merge(additional_species, additional_species_temp, all = TRUE) -->
<!--     counter <- counter + 1 -->
<!--     colnames(bug_list)[counter] <- samp -->

<!--     colnames(additional_species)[counter+1] <- paste('additional_species_',samp,sep='') -->
<!--     } -->
<!--   } -->
<!-- } -->
<!-- bug_list <- data.frame(bug_list, row.names = 1) -->
<!-- additional_species <- data.frame(additional_species, row.names = 1) -->
<!-- colnames(additional_species)[1] <- "NCBI_tax_id" -->
<!-- rm(bug_list_temp) -->
<!-- rm(additional_species_temp) -->

<!-- #Convert NA's to 0 (as they are reads and NA means no reads, aka 0): -->
<!-- #bug_list[is.na(bug_list)] <- 0 -->

<!-- ``` -->

<!-- Merge the additional species per sample in a column (additional_data) and merge it with the bug_list (final_dataframe): -->
<!-- ```{r} -->
<!-- cancel=FALSE -->
<!-- NCBI_tax_id=c() -->
<!-- row_names=c() -->
<!-- additional_taxas=c() -->
<!-- for (i in 1:dim(additional_species)[1]){ -->
<!--   switcher=FALSE -->
<!--   taxa=NA -->
<!--   for (j in 2:dim(additional_species)[2]) { -->
<!--     if (switcher == TRUE && !is.na(additional_species[i,j])) { -->
<!--       if (taxa == additional_species[i,j]){ -->
<!--         #print("they are equal") -->
<!--       } -->
<!--       if (taxa != additional_species[i,j]){ -->
<!--         cancel=TRUE; -->
<!--       } -->
<!--     }else if(!is.na(additional_species[i,j])){ -->
<!--       switcher=TRUE -->
<!--       taxa=additional_species[i,j] -->
<!--     } -->
<!--   } -->
<!--   if (cancel==FALSE){ -->
<!--     rownam=row.names(additional_species[i,]) -->
<!--     row_names=c(row_names, rownam) -->

<!--     taxa_id=additional_species[i,1] -->
<!--     NCBI_tax_id=c(NCBI_tax_id, taxa_id) -->

<!--     additional_taxas=c(additional_taxas, taxa) -->

<!--   }else{print("additional species are not equal")} -->
<!-- } -->

<!-- additional_data=data.frame(row_names, NCBI_tax_id, additional_taxas,row.names = 1) -->

<!-- final_dataframe <- cbind(additional_data,bug_list) -->

<!-- ``` -->

```{r}
# bug_list <- read.csv('E:/JULIOL/DATA/METAHIT/DNA_JOINED/buglist_Methait_DNAjoined.tsv', sep = '\t', row.names = 1 , header = T)
# bug_list <- read.csv('C:/Users/Gerard/Downloads/JULIOL/DATA/METAHIT/DNA_JOINED/buglist_Methait_DNAjoined.tsv', sep = '\t', row.names = 1 , header = T)
bug_list <- read.csv('/mnt/synology/GERARD/DATA/METAHIT/DNA_JOINED/buglist_Methait_DNAjoined.tsv', sep = '\t', row.names = 1 , header = T)

```

Isolate the species from metaphlan output into just_species data frame
```{r}
to_look = "s__"

for (i in row.names(bug_list)) {
  if (grepl(to_look,i)) {
      if (!exists("just_species")){
      just_species <- data.frame(bug_list[i,])
    }
    else if (exists("just_species")){
      just_species <- rbind(just_species, bug_list[i,])
    }
  }
}
just_species[is.na(just_species)] <- 0

# for (i in 1:dim(just_species)[2]) {
#   print(sum(just_species[,i]))
# }
```

```{r}
library(phyloseq)
# rm(taxa_table)
for (i in row.names(just_species)) {
  tax_list <- str_split(i,"\\|", simplify = T)
  kingdom <- str_split(tax_list[1], pattern = "k__", simplify = T)[2]
  phylum<- str_split(tax_list[2], pattern = "p__", simplify = T)[2]
  class<- str_split(tax_list[3], pattern = "c__", simplify = T)[2]
  order<- str_split(tax_list[4], pattern = "o__", simplify = T)[2]
  family<- str_split(tax_list[5], pattern = "f__", simplify = T)[2]
  genus <- str_split(tax_list[6], pattern = "g__", simplify = T)[2]
  specie <- str_split(tax_list[7], pattern = "s__", simplify = T)[2]

  if (!exists("taxa_table")){
      taxa_table <- data.frame(kingdom, phylum, class, order, family, genus, specie, stringsAsFactors = F)
    }
    else if (exists("taxa_table")){
      taxa_table <- rbind(taxa_table, c(kingdom, phylum, class, order, family, genus, specie))
    }
}
# just_species<-just_species/100
for (j in 1:ncol(just_species)) {
  colnames(just_species)[j] <- str_split(colnames(just_species)[j], "_profile", simplify = T)[1]
}
otable_species <- otu_table(just_species, taxa_are_rows = T)
tax = tax_table(taxa_table)
colnames(tax) <- c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species")
taxa_names(tax) <- row.names(just_species)
row.names(taxa_table) <- row.names(just_species)
sampdata <- sample_data(metadata)
phy_species <- phyloseq(otable_species, sampdata, tax)
```

```{r}
for (i in 1:nrow(just_species)) {

  rownames(just_species)[rownames(just_species) == rownames(just_species[i,])] <- taxa_table$specie[rownames(taxa_table) == rownames(just_species[i,])]
}

```

Load CCFA:
```{r}
metadata_CCFA <- read.csv('/mnt/synology/GERARD/DATA/CCFA_project/samples.txt_metadata.tsv', sep = "\t", header = T, row.names = 1) #path to metadata file KATHLEEN
# metadata_CCFA <- read.csv('E:/JULIOL/DATA/samples.txt_metadata.tsv', sep = "\t", header = T, row.names = 1) #path to metadata file KATHLEEN

# metadata_CCFA <- read.csv('C:/Users/Gerard/Downloads/JULIOL/DATA/samples.txt_metadata.tsv', sep = "\t", header = T, row.names = 1) #path to metadata file KATHLEEN

# metadata_CCFA$MOD = "CD"

bug_list_CCFA <- read.csv('/mnt/synology/GERARD/DATA/CCFA_project/buglist_CCFA.tsv', sep = '\t', row.names = 1 , header = T)
# bug_list_CCFA <- read.csv('E:/JULIOL/DATA/buglist_CCFA.tsv', sep = '\t', row.names = 1 , header = T)
# bug_list_CCFA <- read.csv('C:/Users/Gerard/Downloads/JULIOL/DATA/buglist_CCFA.tsv', sep = '\t', row.names = 1 , header = T)

metadata_CCFA$cohort_UC <- "CD"
metadata_CCFA$origin <- "Europe"
metadata_CCFA$country <- "Belgium"
metadata_CCFA$remission_vs_recurrence <- metadata_CCFA$MOD
metadata_CCFA$remission_vs_recurrence[metadata_CCFA$remission_vs_recurrence == "REM"] <- "Remission"
metadata_CCFA$remission_vs_recurrence[metadata_CCFA$remission_vs_recurrence == "REC"] <- "Relapse"
metadata_CCFA$remission_vs_recurrence[metadata_CCFA$TimePoint == "m0"] <- "Relapse"


```


Isolate the species from metaphlan output into just_species data frame
```{r}
to_look = "s__"

for (i in row.names(bug_list_CCFA)) {
  if (grepl(to_look,i)) {
      if (!exists("just_species_CCFA")){
      just_species_CCFA <- data.frame(bug_list_CCFA[i,])
    }
    else if (exists("just_species_CCFA")){
      just_species_CCFA <- rbind(just_species_CCFA, bug_list_CCFA[i,])
    }
  }
}
just_species_CCFA[is.na(just_species_CCFA)] <- 0

# for (i in 1:dim(just_species_CCFA)[2]) {
#   print(sum(just_species_CCFA[,i]))
# }
```

```{r}
# rm(taxa_table)
for (i in row.names(just_species_CCFA)) {
  tax_list <- str_split(i,"\\|", simplify = T)
  kingdom <- str_split(tax_list[1], pattern = "k__", simplify = T)[2]
  phylum<- str_split(tax_list[2], pattern = "p__", simplify = T)[2]
  class<- str_split(tax_list[3], pattern = "c__", simplify = T)[2]
  order<- str_split(tax_list[4], pattern = "o__", simplify = T)[2]
  family<- str_split(tax_list[5], pattern = "f__", simplify = T)[2]
  genus <- str_split(tax_list[6], pattern = "g__", simplify = T)[2]
  specie <- str_split(tax_list[7], pattern = "s__", simplify = T)[2]

  if (!exists("taxa_table_CCFA")){
      taxa_table_CCFA <- data.frame(kingdom, phylum, class, order, family, genus, specie, stringsAsFactors = F)
    }
    else if (exists("taxa_table_CCFA")){
      taxa_table_CCFA <- rbind(taxa_table_CCFA, c(kingdom, phylum, class, order, family, genus, specie))
    }
}
# just_species<-just_species/100
for (j in 1:ncol(just_species_CCFA)) {
  colnames(just_species_CCFA)[j] <- str_split(colnames(just_species_CCFA)[j], "_profile", simplify = T)[1]
}
otable_species_CCFA <- otu_table(just_species_CCFA, taxa_are_rows = T)
tax_CCFA = tax_table(taxa_table_CCFA)
colnames(tax_CCFA) <- c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species")
taxa_names(tax_CCFA) <- row.names(just_species_CCFA)
row.names(taxa_table_CCFA) <- row.names(just_species_CCFA)
sampdata_CCFA <- sample_data(metadata_CCFA)
phy_species_CCFA <- phyloseq(otable_species_CCFA, sampdata_CCFA, tax_CCFA)
```

```{r}
for (i in 1:nrow(just_species_CCFA)) {

  rownames(just_species_CCFA)[rownames(just_species_CCFA) == rownames(just_species_CCFA[i,])] <- taxa_table_CCFA$specie[rownames(taxa_table_CCFA) == rownames(just_species_CCFA[i,])]
}

```


Load hmp metadata and functional profiles:
```{r}
metadata_usa <- read.csv('/mnt/synology/GERARD/DATA/HUTTENHOWER_LAB/hmp2_metadata.csv', sep = ",", header = T) #METAHIT ORIGINALS
# metadata_usa <- read.csv('E:/JULIOL/DATA/hmp2_metadata.csv', sep = ",", header = T) #METAHIT ORIGINALS
# metadata_usa <- read.csv('C:/Users/Gerard/Downloads/JULIOL/DATA/hmp2_metadata.csv', sep = ",", header = T) #METAHIT ORIGINALS

metadata_usa <- metadata_usa[metadata_usa$data_type == 'metagenomics',]

rownames(metadata_usa) <- metadata_usa$External.ID
metadata_usa$cohort_UC <- metadata_usa$diagnosis
metadata_usa$origin <- "America"
metadata_usa$country <- "USA"

metadata_usa$cohort_UC[metadata_usa$cohort_UC == "nonIBD"] <- "healthy"
bug_list_usa <- read.csv('/mnt/synology/GERARD/DATA/HUTTENHOWER_LAB/hmp2_mgx_taxonomy.tsv', sep = '\t', row.names = 1 , header = T)
# bug_list_usa <- read.csv('E:/JULIOL/DATA/hmp2_mgx_taxonomy.tsv', sep = '\t', row.names = 1 , header = T)
# bug_list_usa <- read.csv('C:/Users/Gerard/Downloads/JULIOL/DATA/hmp2_mgx_taxonomy.tsv', sep = '\t', row.names = 1 , header = T)

```

Isolate the species from metaphlan output into just_species data frame
```{r}
to_look = "s__"

for (i in row.names(bug_list_usa)) {
  if (grepl(to_look,i)) {
      if (!exists("just_species_usa")){
      just_species_usa <- data.frame(bug_list_usa[i,])
    }
    else if (exists("just_species_usa")){
      just_species_usa <- rbind(just_species_usa, bug_list_usa[i,])
    }
  }
}
just_species_usa[is.na(just_species_usa)] <- 0
just_species_usa <- just_species_usa[!colSums(just_species_usa) == 0]
# for (i in 1:dim(just_species_usa)[2]) {
#   print(sum(just_species_usa[,i]))
# }
```

```{r}
# rm(taxa_table)
for (i in row.names(just_species_usa)) {
  tax_list <- str_split(i,"\\|", simplify = T)
  kingdom <- str_split(tax_list[1], pattern = "k__", simplify = T)[2]
  phylum<- str_split(tax_list[2], pattern = "p__", simplify = T)[2]
  class<- str_split(tax_list[3], pattern = "c__", simplify = T)[2]
  order<- str_split(tax_list[4], pattern = "o__", simplify = T)[2]
  family<- str_split(tax_list[5], pattern = "f__", simplify = T)[2]
  genus <- str_split(tax_list[6], pattern = "g__", simplify = T)[2]
  specie <- str_split(tax_list[7], pattern = "s__", simplify = T)[2]

  if (!exists("taxa_table_usa")){
      taxa_table_usa <- data.frame(kingdom, phylum, class, order, family, genus, specie, stringsAsFactors = F)
    }
    else if (exists("taxa_table_usa")){
      taxa_table_usa <- rbind(taxa_table_usa, c(kingdom, phylum, class, order, family, genus, specie))
    }
}
# just_species<-just_species/100
for (j in 1:ncol(just_species_usa)) {
  colnames(just_species_usa)[j] <- str_split(colnames(just_species_usa)[j], "_profile", simplify = T)[1]
}
otable_species_usa <- otu_table(just_species_usa, taxa_are_rows = T)
tax_usa = tax_table(taxa_table_usa)
colnames(tax_usa) <- c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species")
taxa_names(tax_usa) <- row.names(just_species_usa)
row.names(taxa_table_usa) <- row.names(just_species_usa)
sampdata_usa <- sample_data(metadata_usa)
phy_species_usa <- phyloseq(otable_species_usa, sampdata_usa, tax_usa)
```

```{r}
for (i in 1:nrow(just_species_usa)) {

  rownames(just_species_usa)[rownames(just_species_usa) == rownames(just_species_usa[i,])] <- taxa_table_usa$specie[rownames(taxa_table_usa) == rownames(just_species_usa[i,])]
}

```

Make PCoA:
```{r}
library(ggplot2)

phy_species_europe <- merge_phyloseq(phy_species, phy_species_CCFA)

phylo_ord <- ordinate(phy_species_europe, method = "PCoA", distance = 'bray')
p1 <- plot_ordination(phy_species_europe, phylo_ord, color = "country", shape = "cohort_UC")
# p1 <- plot_ordination(phy_species_usa, phylo_ord, color = "cohort_UC", shape = "origin")
p1 + theme_classic() +
    scale_color_brewer(palette = "Dark2") +
    # geom_text(aes(label=colnames(otu_table(phyloseq_obj)), hjust= 0, vjust = 2)) +
    ggtitle("PCoA Bray Curtis Spanish and Belgian Cohorts") +
    theme(plot.title = element_text(hjust = 0.5))


# phy_species_all <- merge_phyloseq(phy_species, phy_species_CCFA, phy_species_usa)
phy_species_all <- merge_phyloseq(phy_species, phy_species_CCFA, phy_species_usa)

phylo_ord <- ordinate(phy_species_all, method = "PCoA", distance = 'bray')
p1 <- plot_ordination(phy_species_all, phylo_ord, color = "origin", shape = "cohort_UC")
# p1 <- plot_ordination(phy_species_usa, phylo_ord, color = "cohort_UC", shape = "origin")
p1 + theme_classic() +
    scale_color_brewer(palette = "Dark2") +
    # geom_text(aes(label=colnames(otu_table(phyloseq_obj)), hjust= 0, vjust = 2)) +
    ggtitle("Bug list clustering") +
    theme(plot.title = element_text(hjust = 0.5))


```


```{r}
library(vegan)

# CDs <- subset_samples(phy_species_all, cohort_UC == "CD")
CDs <- subset_samples(phy_species_all, cohort_UC == "CD" & origin == "Europe")

CDs_species <- data.frame(otu_table(CDs))
CDs_metadata <- data.frame(sample_data(CDs))

bl <- as.data.frame(t(CDs_species))

adonis_res <- adonis2(bl ~ country, data = CDs_metadata)

adonis_res$variable = rownames(adonis_res)
adonis_res

CDs <- subset_samples(phy_species_all, cohort_UC == "CD")

CDs_species <- data.frame(otu_table(CDs))
CDs_metadata <- data.frame(sample_data(CDs))

bl <- as.data.frame(t(CDs_species))

adonis_res <- adonis2(bl ~ origin, data = CDs_metadata)

adonis_res$variable = rownames(adonis_res)
adonis_res

UCs <- subset_samples(phy_species_all, cohort_UC == "UC")

UCs_species <- data.frame(otu_table(UCs))
UCs_metadata <- data.frame(sample_data(UCs))

bl <- as.data.frame(t(UCs_species))

adonis_res <- adonis2(bl ~ origin, data = UCs_metadata)

adonis_res$variable = rownames(adonis_res)
adonis_res

healthies <- subset_samples(phy_species_all, cohort_UC == "CD")

healthies_species <- data.frame(otu_table(healthies))
healthies_metadata <- data.frame(sample_data(healthies))

bl <- as.data.frame(t(healthies_species))

adonis_res <- adonis2(bl ~ origin, data = healthies_metadata)

adonis_res$variable = rownames(adonis_res)
adonis_res

```
Import metaphlan tree:
```{r}
# tree <- read_tree('~/Downloads/mpa_v30_CHOCOPhlAn_201901_species_tree.nwk')
# tree <- read_tree('E:/JULIOL/DATA/mpa_v30_CHOCOPhlAn_201901_species_tree.nwk')
tree <- read_tree('C:/Users/Gerard/Downloads/JULIOL/DATA/mpa_v30_CHOCOPhlAn_201901_species_tree.nwk')

tree$tip.label <- sub("GCA_[0-9]+\\|", "", tree$tip.label)
tree <- phy_tree(tree)

# phy_species <- phyloseq(otable_species, sampdata, tax, tree)
#
# plot_tree(phy_species, color="MOD") #, shape = "remission_vs_recurrence")
```


Europe remission and actives comparison:
```{r}
europe_remission <- subset_samples(phy_species_all, country == "Belgium" & remission_vs_recurrence == "Remission" & TimePoint == "m6")

europe_remission_species <- data.frame(otu_table(europe_remission))
europe_remission_metadata <- data.frame(sample_data(europe_remission))

europe_remission <- subset_samples(phy_species_all, country == "Spain" & remission_vs_recurrence == "Remission" & timep == "0" & cohort_UC == "CD")

europe_remission_species <- cbind(europe_remission_species, data.frame(otu_table(europe_remission)))
europe_remission_metadata <- rbind(europe_remission_metadata, data.frame(sample_data(europe_remission)))

phy_europe_remission <- phyloseq(otu_table(europe_remission_species, taxa_are_rows = T), sample_data(europe_remission_metadata), tax, tree)

plot_tree(phy_europe_remission, color= "country")

# bl <- as.data.frame(t(europe_remission_species))
d <- UniFrac(phy_europe_remission, weighted=T)

# adonis_res <- adonis2(bl ~ country, data = europe_remission_metadata)

adonis_res <- adonis2(d ~ country, data = europe_remission_metadata)

adonis_res$variable = rownames(adonis_res)
adonis_res



europe_relapse <- subset_samples(phy_species_all, country == "Belgium" & remission_vs_recurrence == "Relapse")
europe_relapse_species <- data.frame(otu_table(europe_relapse))
europe_relapse_metadata <- data.frame(sample_data(europe_relapse))

# europe_relapse <- subset_samples(phy_species_all, country == "Belgium" & remission_vs_recurrence == "Remission" & TimePoint == "m0")
#
# europe_relapse_species <- cbind(europe_relapse_species, data.frame(otu_table(europe_relapse)))
# europe_relapse_metadata <- rbind(europe_relapse_metadata, data.frame(sample_data(europe_relapse)))

europe_relapse <- subset_samples(phy_species_all, country == "Spain" & remission_vs_recurrence == "Relapse" & cohort_UC == "CD")

europe_relapse_species <- cbind(europe_relapse_species, data.frame(otu_table(europe_relapse)))
europe_relapse_metadata <- rbind(europe_relapse_metadata, data.frame(sample_data(europe_relapse)))

phy_europe_relapse <- phyloseq(otu_table(europe_relapse_species, taxa_are_rows = T), sample_data(europe_relapse_metadata), tax, tree)

plot_tree(phy_europe_remission, color= "country")

# bl <- as.data.frame(t(europe_relapse_species))
d <- UniFrac(phy_europe_relapse, weighted=T)


# adonis_res <- adonis2(bl ~ country, data = europe_relapse_metadata)
adonis_res <- adonis2(d ~ country, data = europe_relapse_metadata)

adonis_res$variable = rownames(adonis_res)
adonis_res


```
Tests Unweighted UniFrac:
```{r}
d <- UniFrac(phy_europe_remission, weighted=T)

# adonis_res <- adonis2(bl ~ country, data = europe_remission_metadata)

adonis_res <- adonis2(d ~ country, data = europe_remission_metadata)

adonis_res$variable = rownames(adonis_res)
adonis_res

d <- UniFrac(phy_europe_relapse, weighted=F)


# adonis_res <- adonis2(bl ~ country, data = europe_relapse_metadata)
adonis_res <- adonis2(d ~ country, data = europe_relapse_metadata)

adonis_res$variable = rownames(adonis_res)
adonis_res

```

PCoA Spanish and Belgian
```{r}
phy_species_europe <- merge_phyloseq(phy_species_europe, tree)
d <- UniFrac(phy_species_europe, weighted=T)


phylo_ord <- ordinate(phy_species_europe, method = "PCoA", distance = d)
p1 <- plot_ordination(phy_species_europe, phylo_ord, color = "country", shape = "cohort_UC")
# p1 <- plot_ordination(phy_species_usa, phylo_ord, color = "cohort_UC", shape = "origin")
p1 + theme_classic() +
    scale_color_brewer(palette = "Dark2") +
    # geom_text(aes(label=colnames(otu_table(phyloseq_obj)), hjust= 0, vjust = 2)) +
    ggtitle("PCoA weighted UniFrac Spanish and Belgian Cohorts") +
    theme(plot.title = element_text(hjust = 0.5))


d <- UniFrac(phy_species_europe, weighted=F)


phylo_ord <- ordinate(phy_species_europe, method = "PCoA", distance = d)
p1 <- plot_ordination(phy_species_europe, phylo_ord, color = "country", shape = "cohort_UC")
# p1 <- plot_ordination(phy_species_usa, phylo_ord, color = "cohort_UC", shape = "origin")
p1 + theme_classic() +
    scale_color_brewer(palette = "Dark2") +
    # geom_text(aes(label=colnames(otu_table(phyloseq_obj)), hjust= 0, vjust = 2)) +
    ggtitle("PCoA unweighted UniFrac Spanish and Belgian Cohorts") +
    theme(plot.title = element_text(hjust = 0.5))
```
